src/metpy/plots/station_plot.py

```diff
@@ -371,13 +371,9 @@ def _make_kwargs(self, kwargs):
     def _to_string_list(vals, fmt):
         """Convert a sequence of values to a list of strings."""
         if fmt is None:
-            import sys
-            if sys.version_info >= (3, 11):
-                fmt = 'z.0f'
-            else:
-                def fmt(s):
-                    """Perform default formatting with no decimal places and no negative 0."""
-                    return format(round(s, 0) + 0., '.0f')
+            def fmt(s):
+                """Perform default formatting with no decimal places and no negative 0."""
+                return format(round(s, 0) + 0., '.0f')
```

Comment:
Actually, instead the format string should have been kept (apparently this got swallowed when I reviewed earlier):
```suggestion
            fmt = 'z.0f'
```